PR: https://github.com/odoo/odoo/pull/232873

From: 044a6cf0e0a25cebab20567415ef498de31d06f5
From: hamza
Date: 2025-11-17 13:42:40

Breaking data model changes scores: del:3 + add:0, change matches:
-    mobility_card = fields.Char(groups="fleet.fleet_group_user")
-    mobility_card = fields.Char(readonly=True)
-    mobility_card = fields.Char(compute='_compute_mobility_card', store=True)

Total Changes: 19

[IMP] hr_fleet : making Fleet mobility card exclusive to BE

* Removed the field from the employee general view , and put it in the employee view of the be localisation to make it exclusive.
* made the field only available to employees of belgian companies in the fleet_vehicle view.

task - 5176316

closes odoo/odoo#232873

Related: odoo/enterprise#97986
Related: odoo/upgrade#8733
Signed-off-by: Yannick Tivisse (yti) <yti@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_fleet/models/employee.py
+++ b/addons/hr_fleet/models/employee.py
@@ -14,7 +14,6 @@ class HrEmployee(models.Model):
         groups="fleet.fleet_group_manager,hr.group_hr_user",
     )
     license_plate = fields.Char(compute="_compute_license_plate", search="_search_license_plate", groups="hr.group_hr_user")
-    mobility_card = fields.Char(groups="fleet.fleet_group_user")
 
     def action_open_employee_cars(self):
         self.ensure_one()
@@ -81,11 +80,6 @@ class HrEmployee(models.Model):
                             'future_driver_id': vals['work_contact_id'],
                         })
 
-        if 'mobility_card' in vals:
-            car_ids = self.env['fleet.vehicle'].sudo().search([
-                ('driver_employee_id', 'in', self.ids),
-            ])
-            car_ids._compute_mobility_card()
         return res
 
 
@@ -93,7 +87,6 @@ class HrEmployeePublic(models.Model):
     _inherit = 'hr.employee.public'
 
     employee_cars_count = fields.Integer(compute="_compute_employee_cars_count", string="Cars", groups="fleet.fleet_group_user")
-    mobility_card = fields.Char(readonly=True)
 
     def action_open_employee_cars(self):
         return self.env['hr.employee'].browse(self.ids).action_open_employee_cars()

--- a/addons/hr_fleet/models/fleet_vehicle.py
+++ b/addons/hr_fleet/models/fleet_vehicle.py
@@ -7,7 +7,6 @@ from odoo.exceptions import ValidationError
 class FleetVehicle(models.Model):
     _inherit = 'fleet.vehicle'
 
-    mobility_card = fields.Char(compute='_compute_mobility_card', store=True)
     driver_employee_id = fields.Many2one(
         'hr.employee', 'Driver (Employee)',
         compute='_compute_driver_employee_id', store=True,
@@ -15,6 +14,7 @@ class FleetVehicle(models.Model):
         tracking=True,
         index='btree_not_null',
     )
+
     driver_employee_name = fields.Char(related="driver_employee_id.name")
     future_driver_employee_id = fields.Many2one(
         'hr.employee', 'Future Driver (Employee)',
@@ -51,16 +51,6 @@ class FleetVehicle(models.Model):
             employees = employees_by_partner_id_and_company_id.get((vehicle.future_driver_id, vehicle.company_id))
             vehicle.future_driver_employee_id = employees[0] if employees else False
 
-    @api.depends('driver_id')
-    def _compute_mobility_card(self):
-        for vehicle in self:
-            employee = self.env['hr.employee']
-            if vehicle.driver_id:
-                employee = employee.search([('work_contact_id', '=', vehicle.driver_id.id)], limit=1)
-                if not employee:
-                    employee = employee.search([('user_id.partner_id', '=', vehicle.driver_id.id)], limit=1)
-            vehicle.mobility_card = employee.mobility_card
-
     def _update_create_write_vals(self, vals):
         if 'driver_employee_id' in vals:
             partner = False
